plutus-playground-client/test/Data/Foldable/ExtraTests.purs

```diff
@@ -37,3 +39,20 @@ interleaveTests = do
       equal
         [ 1, 0, 2, 0, 3 ]
         (interleave 0 [ 1, 2, 3 ])
+
+countConsecutiveTests :: TestSuite
+countConsecutiveTests = do
+  suite "countConsecutive" do
+    test "Empty." do
+      equal
+        (countConsecutive ([] :: Array String))
+        []
+    test "Concrete example." do
+      equal
+        ( countConsecutive
+            [ "Foo", "Foo", "Foo", "Foo", "Foo", "Bar", "Bar", "Foo", "Foo", "Foo" ]
+        )
+        [ 5 /\ "Foo"
+        , 2 /\ "Bar"
+        , 3 /\ "Foo"
+        ]
```

Comment:
maybe some quickchecks...

Reply:
Yes, good call. We can actually test this really easily by reconstructing the original sequence:

``` purescript
quickCheck do
  msgs :: Array Msg <- arbitrary
  pure $ msgs == foldMap (uncurry replicate) (countConsecutive msgs)
```

👍